docs/life-of-a-feature.md

```diff
@@ -0,0 +1,29 @@
+# Life of a Feature
+
+## Do's
+
+1. Understand the goal by answering "what's the problem we're solving?".
+1. Write technical design docs.
+    - Do some researches and provide background information and overview.
+    - Find a simple and sustainable solution. Technical decisions usually come along with trade-offs among several options. We need a simple solution for a difficult problem. The solution should also be scalable to future changes or growth if possible.
+    - Split a large design into smaller ones, e.g. backend, UI, UX, etc.
+    - ```(IMPORTANT)``` Define data model and design database schema. According to our [Version Management](version-management.md), any schema change could cause disruption to existing customers. Uncareful design will also lead to software scalability issues.
+    - Design API by following [API style guide](https://github.com/bytebase/bytebase/blob/main/docs/api-style-guide.md)
+    - Put thoughts on naming, because it's hard to change the names in database schemas and APIs. We can have different names for technical pieces and products.
+    - Collabrate with peers and tech leads.
+    - Write doc including comments in English. We are doing open-source with contributors globally.
+2. Coding.
+    - Follow [code review guide](code-review-guide.md) (small changes, effective communication, collabrate, and ```respect```).
+    - Split changes to database schema, API, backend, frontend if possible, because you will get different reviewers looking at different parts. For example, we can make backend do dual writes for any API changes, switch the reads on UI, and clean up the dual writes in bacnkend.
+    - ```(IMPORTANT)``` Think about compatability and don't break existing users. This usually happen if we change database schemas or APIs.
+    - Guard new features behind a release flag especially for frontend. We should release the feature only when it's mature.
```

Comment:
What's the best practice for doing this now?

Reply:
Done. Added an example for isDev().